library/src/main/java/com/prolificinteractive/materialcalendarview/MaterialCalendarView.java

```diff
@@ -65,6 +68,8 @@
     private CalendarDay currentMonth;
     private TitleFormatter titleFormatter = DEFAULT_TITLE_FORMATTER;
 
+    private Map<String,DayViewDecorator> dayViewDecorators;
```

Comment:
Instead of having a Map, what if we had each decorator responsible for which dates it wanted to decorate? Then we could have multiple decorators act on the same day. Something like:

```java
boolean shouldDecorate(CalendarDay date);
```